crates/bevy_ecs/src/query/state.rs

```diff
@@ -538,6 +678,28 @@ impl<Q: WorldQuery, F: ReadOnlyWorldQuery> QueryState<Q, F> {
         Ok(values.map(|x| x.assume_init()))
     }
 
+    /// Gets the read-only query results for the given [`World`] and array of [`Entity`], where the last change and
+    /// the current change tick are given.
+    ///
+    /// # Panics
+    ///
+    /// Panics if any item cannot be retrieved.
+    ///
+    /// # Safety
+    ///
+    /// This must be called on the same `World` that the `Query` was generated from:
+    /// use `QueryState::validate_world` to verify this.
+    pub(crate) unsafe fn many_read_only_manual<'w, const N: usize>(
+        &self,
+        world: &'w World,
+        entities: [Entity; N],
+        last_run: Tick,
+        this_run: Tick,
+    ) -> [ROQueryItem<'w, Q>; N] {
+        self.get_many_read_only_manual(world, entities, last_run, this_run)
+            .unwrap()
```

Comment:
Using `unwrap` won't have a good error message, since the error's `Debug` implementation will be used. It'd be better to use a custom panic message or the `Display` impl.

Reply:
I've updated the PR to use a `match` statement to extract the error, and then `panic!` with a short context message, and then the `Display` of the inner error.

```rust
match self.get_many_unchecked_manual(world, entities, last_run, this_run) {
    Ok(items) => items,
    Err(error) => panic!("Cannot get query result: {error}"),
}
```
I noticed there were multiple places where the above pattern was appropriate, so I've implemented it in all such locations within `state.rs`. I also moved `QuerySingleError` into the new `error.rs` module for consistency.